drivers/SmartThings/samsung-audio/src/handlers.lua

```diff
@@ -80,33 +80,59 @@ end
 
 function CapabilityHandlers.handle_mute(driver, device, cmd)
   local ip = device:get_field("ip")
-  command.mute(ip)
+  local muteStatus = command.mute(ip)
+  if muteStatus then
+    if muteStatus.muted ~= "off" then
+      device:emit_event(capabilities.audioMute.mute.muted())
+    else
+      device:emit_event(capabilities.audioMute.mute.unmuted())
+    end
+  end
 end
 
 function CapabilityHandlers.handle_unmute(driver, device, cmd)
   local ip = device:get_field("ip")
-  command.unmute(ip)
+  local muteStatus = command.unmute(ip)
+  if muteStatus then
+    if muteStatus.muted ~= "off" then
+      device:emit_event(capabilities.audioMute.mute.muted())
+    else
+      device:emit_event(capabilities.audioMute.mute.unmuted())
+    end
+  end
 end
 
 function CapabilityHandlers.handle_volume_up(driver, device, cmd)
   local ip = device:get_field("ip")
   local vol = command.volume(ip)
   if vol then
-    command.set_volume(ip, vol.volume + 5)
+    local set_vol = command.set_volume(ip, tonumber(vol.volume) + 5)
```

Comment:
sorry, nit, but can you make this volume delta a constant defined at the top of the file?

`local VOLUME_STEP = 5` or something like that?

Reply:
> sorry, nit, but can you make this volume delta a constant defined at the top of the file?
> 
> `local VOLUME_STEP = 5` or something like that?

Done.